pandas/core/arrays/arrow/array.py

```diff
@@ -1541,21 +1513,94 @@ def pyarrow_meth(data, skip_nulls, **kwargs):
+            std, var, sem, kurt, skew }.
+
+        Returns
+        -------
+        pa.ChunkedArray
+        """
+        pa_type = self._pa_array.type
+
+        if name in ["any", "all"] and (
+            pa.types.is_integer(pa_type)
+            or pa.types.is_floating(pa_type)
+            or pa.types.is_duration(pa_type)
+            or pa.types.is_decimal(pa_type)
+        ):
+            # pyarrow only supports any/all for boolean dtype, we allow
+            #  for other dtypes, matching our non-pyarrow behavior
+            if pa.types.is_duration(pa_type):
+                data_to_cmp = self._pa_array.cast(pa.int64())
+            else:
+                data_to_cmp = self._pa_array
+
+            not_eq = pc.not_equal(data_to_cmp, 0)
+            data_to_reduce = not_eq
+
+        elif name in ["min", "max", "sum"] and pa.types.is_duration(pa_type):
+            data_to_reduce = self._pa_array.cast(pa.int64())
+
+        elif name in ["median", "mean", "std", "sem"] and pa.types.is_temporal(pa_type):
+            nbits = pa_type.bit_width
+            if nbits == 32:
+                data_to_reduce = self._pa_array.cast(pa.int32())
+            else:
+                data_to_reduce = self._pa_array.cast(pa.int64())
+
+        elif name in ["sem", "std", "var"] and pa.types.is_boolean(pa_type):
+            data_to_reduce = self._pa_array.cast(pa.uint8())
+
+        else:
+            data_to_reduce = self._pa_array
+
+        return data_to_reduce
+
+    def _maybe_cast_reduction_result(
+        self, result: pa.Scalar | pa.ChunkedArray, name: str
+    ):
+        """
+        Maybe cast a reduction result to a different pyarrow type.
+
+        See ArrowExtensionArray._values_for_reduction.
+
+        Parameters
+        ----------
+        result : pa.Scalar | pa.ChunkedArray
+        name : str
+            Name of the function, supported values are:
+            { any, all, min, max, sum, mean, median, prod,
+            std, var, sem, kurt, skew }.
+
+        Returns
+        -------
+        pa.Scalar or pa.ChunkedArray
+        """
+        pa_type = self._pa_array.type
         if name in ["min", "max", "sum"] and pa.types.is_duration(pa_type):
             result = result.cast(pa_type)
         if name in ["median", "mean"] and pa.types.is_temporal(pa_type):
             if not pa_version_under13p0:
                 nbits = pa_type.bit_width
                 if nbits == 32:
-                    result = result.cast(pa.int32(), **cast_kwargs)
+                    result = pc.cast(result, pa.int32(), safe=False)
                 else:
-                    result = result.cast(pa.int64(), **cast_kwargs)
+                    result = pc.cast(result, pa.int64(), safe=False)
             result = result.cast(pa_type)
         if name in ["std", "sem"] and pa.types.is_temporal(pa_type):
-            result = result.cast(pa.int64(), **cast_kwargs)
+            result = pc.cast(result, pa.int64(), safe=False)
```

Comment:
for my edification: is this materially different or just idiomatic?

Reply:
No difference at all. The method didn't support the safe keyword until pyarrow version 8 whereas the function supported it all along. Switching to the function was just to remove an xfail.